## Basic Imports

### Problem 3

In [66]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### Problem 4

In [67]:
%matplotlib inline

## Create dataframes

### Problem 5

In [68]:
gdp_df = pd.read_csv('C:/Users/jrior/Documents/NSS_DATA_Analytics/projects/gdp-and-internet-usage-jrioross/data/gdp_percapita.csv')

In [69]:
gdp_df.head(n = 6)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2018  2190.240321              NaN
1     Afghanistan  2017  2202.570851              NaN
2     Afghanistan  2016  2200.764487              NaN
3     Afghanistan  2015  2212.750899              NaN
4     Afghanistan  2014  2249.253738              NaN
5     Afghanistan  2013  2264.317575              NaN

### Problem 6

In [70]:
internet_df = pd.read_csv('C:/Users/jrior/Documents/NSS_DATA_Analytics/projects/gdp-and-internet-usage-jrioross/data/internet_use.csv', nrows=4496)

In [71]:
internet_df.head(n = 6)

Country or Area  Year     Value  Value Footnotes
0     Afghanistan  2014  6.390000              NaN
1     Afghanistan  2013  5.900000              NaN
2     Afghanistan  2012  5.454545              NaN
3     Afghanistan  2011  5.000000              NaN
4     Afghanistan  2010  4.000000              NaN
5     Afghanistan  2009  3.550000              NaN

### Problem 7

In [72]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6608 entries, 0 to 6607
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  6608 non-null   object 
 1   Year             6608 non-null   object 
 2   Value            6606 non-null   float64
 3   Value Footnotes  1 non-null      float64
dtypes: float64(2), object(2)
memory usage: 206.6+ KB


In [73]:
internet_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4496 entries, 0 to 4495
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  4496 non-null   object 
 1   Year             4496 non-null   object 
 2   Value            4495 non-null   float64
 3   Value Footnotes  948 non-null    float64
dtypes: float64(2), object(2)
memory usage: 140.6+ KB


**gdp_df has 4 columns and 6,608 rows
internet_df has 4 columns and 4,495 rows**

### Problem 8

**Country or Area: object  
Year: object  
Value: float64  
Value Footnotes: float64**

### Problem 9

In [74]:
gdp_df.tail(10)

Country or Area                                      Year        Value  \
6598        Zimbabwe                                      1997  3580.048793   
6599        Zimbabwe                                      1996  3520.430146   
6600        Zimbabwe                                      1995  3226.413930   
6601        Zimbabwe                                      1994  3263.934978   
6602        Zimbabwe                                      1993  3033.504852   
6603        Zimbabwe                                      1992  3054.889178   
6604        Zimbabwe                                      1991  3426.598094   
6605        Zimbabwe                                      1990  3324.348171   
6606   footnoteSeqID                                  Footnote          NaN   
6607               2  Excludes South Sudan after July 9, 2011.          NaN   

      Value Footnotes  
6598              NaN  
6599              NaN  
6600              NaN  
6601              NaN  
6602              NaN  
6603              NaN  
6604              NaN  
6605              NaN  
6606              NaN  
6607              NaN

In [75]:
internet_df.tail(10)

Country or Area      Year     Value  Value Footnotes
4486        Zimbabwe      2001  0.799846              NaN
4487        Zimbabwe      2000  0.401434              NaN
4488        Zimbabwe      1999  0.161676              NaN
4489        Zimbabwe      1998  0.081648              NaN
4490        Zimbabwe      1997  0.033080              NaN
4491        Zimbabwe      1996  0.016790              NaN
4492        Zimbabwe      1995  0.007684              NaN
4493        Zimbabwe      1994  0.001739              NaN
4494        Zimbabwe      1990  0.000000              NaN
4495   footnoteSeqID  Footnote       NaN              NaN

### Remove the footnote rows

In [76]:
gdp_df.drop(gdp_df.tail(2).index,inplace=True) # drop last 2 rows

In [77]:
gdp_df.tail(10)

Country or Area  Year        Value  Value Footnotes
6596        Zimbabwe  1999  3600.849291              NaN
6597        Zimbabwe  1998  3653.920016              NaN
6598        Zimbabwe  1997  3580.048793              NaN
6599        Zimbabwe  1996  3520.430146              NaN
6600        Zimbabwe  1995  3226.413930              NaN
6601        Zimbabwe  1994  3263.934978              NaN
6602        Zimbabwe  1993  3033.504852              NaN
6603        Zimbabwe  1992  3054.889178              NaN
6604        Zimbabwe  1991  3426.598094              NaN
6605        Zimbabwe  1990  3324.348171              NaN

In [78]:
internet_df.drop(internet_df.tail(1).index,inplace=True) # drop last row

In [79]:
internet_df.tail(10)

Country or Area  Year     Value  Value Footnotes
4485        Zimbabwe  2002  3.994356              NaN
4486        Zimbabwe  2001  0.799846              NaN
4487        Zimbabwe  2000  0.401434              NaN
4488        Zimbabwe  1999  0.161676              NaN
4489        Zimbabwe  1998  0.081648              NaN
4490        Zimbabwe  1997  0.033080              NaN
4491        Zimbabwe  1996  0.016790              NaN
4492        Zimbabwe  1995  0.007684              NaN
4493        Zimbabwe  1994  0.001739              NaN
4494        Zimbabwe  1990  0.000000              NaN

### Problem 10

In [80]:
gdp_df.drop(columns = ['Value Footnotes'],inplace=True)

In [81]:
gdp_df.tail()

Country or Area  Year        Value
6601        Zimbabwe  1994  3263.934978
6602        Zimbabwe  1993  3033.504852
6603        Zimbabwe  1992  3054.889178
6604        Zimbabwe  1991  3426.598094
6605        Zimbabwe  1990  3324.348171

In [82]:
internet_df.drop(columns = ['Value Footnotes'],inplace=True)

In [83]:
internet_df.tail()

Country or Area  Year     Value
4490        Zimbabwe  1997  0.033080
4491        Zimbabwe  1996  0.016790
4492        Zimbabwe  1995  0.007684
4493        Zimbabwe  1994  0.001739
4494        Zimbabwe  1990  0.000000

### Problems 11 & 12

In [87]:
gdp_df = gdp_df.rename(columns = {'Country or Area' : 'Country', 'Value' : 'GDP_Per_Capita'})

In [88]:
internet_df = internet_df.rename(columns = {'Country or Area' : 'Country', 'Value' : 'Internet_Users_Pct'})

### Problem 13

In [95]:
gdp_and_internet_use = gdp_df.merge(internet_df, how = 'outer')

### Problem 14 & 15

**Checking specific country to ensure correct merge**

In [100]:
gdp_and_internet_use[gdp_and_internet_use.Country == 'Afghanistan']

Country  Year  GDP_Per_Capita  Internet_Users_Pct
0     Afghanistan  2018     2190.240321                 NaN
1     Afghanistan  2017     2202.570851                 NaN
2     Afghanistan  2016     2200.764487                 NaN
3     Afghanistan  2015     2212.750899                 NaN
4     Afghanistan  2014     2249.253738            6.390000
5     Afghanistan  2013     2264.317575            5.900000
6     Afghanistan  2012     2220.481526            5.454545
7     Afghanistan  2011     2037.608607            5.000000
8     Afghanistan  2010     2093.743565            4.000000
9     Afghanistan  2009     1881.777626            3.550000
10    Afghanistan  2008     1587.792070            1.840000
11    Afghanistan  2007     1562.876516            1.900000
12    Afghanistan  2006     1407.708041            2.107124
13    Afghanistan  2005     1376.686376            1.224148
14    Afghanistan  2004     1284.127338            0.105809
15    Afghanistan  2003     1322.141328            0.087891
16    Afghanistan  2002     1272.900914            0.004561
6606  Afghanistan  2001             NaN            0.004723
6607  Afghanistan  1990             NaN            0.000000

In [101]:
gdp_and_internet_use.head()

Country  Year  GDP_Per_Capita  Internet_Users_Pct
0  Afghanistan  2018     2190.240321                 NaN
1  Afghanistan  2017     2202.570851                 NaN
2  Afghanistan  2016     2200.764487                 NaN
3  Afghanistan  2015     2212.750899                 NaN
4  Afghanistan  2014     2249.253738                6.39

In [102]:
gdp_and_internet_use.tail()

Country  Year  GDP_Per_Capita  Internet_Users_Pct
7659   Yemen  1999             NaN            0.056629
7660   Yemen  1998             NaN            0.023323
7661   Yemen  1997             NaN            0.015025
7662   Yemen  1996             NaN            0.000621
7663   Yemen  1990             NaN            0.000000

### Problem 16

In [107]:
gdp_and_internet_use[gdp_and_internet_use['Year'].isin(['2004', '2009', '2014'])]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
4               Afghanistan  2014     2249.253738            6.390000
9               Afghanistan  2009     1881.777626            3.550000
14              Afghanistan  2004     1284.127338            0.105809
21                  Albania  2014    11828.232264           60.100000
26                  Albania  2009    10528.778102           41.200000
...                     ...   ...             ...                 ...
7627  Virgin Islands (U.S.)  2009             NaN           27.396510
7632  Virgin Islands (U.S.)  2004             NaN           27.377009
7644                  Yemen  2014             NaN           22.550000
7649                  Yemen  2009             NaN            9.960000
7654                  Yemen  2004             NaN            0.881223

[830 rows x 4 columns]

### Problem 17

In [109]:
gdp_int_2004 = gdp_and_internet_use[gdp_and_internet_use.Year == '2004']
gdp_int_2009 = gdp_and_internet_use[gdp_and_internet_use.Year == '2009']
gdp_int_2014 = gdp_and_internet_use[gdp_and_internet_use.Year == '2014']

### Problem 18

gdp_int_2014[gdp_int_2014.Internet_Users_Pct == max(gdp_int_2014.Internet_Users_Pct)]

**Max Internet Users Percentage in 2014: Iceland with 98.16%**

### Problem 19

In [115]:
gdp_int_2014[gdp_int_2014.Internet_Users_Pct == min(gdp_int_2014.Internet_Users_Pct)]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
6838  Dem. People's Rep. of Korea  2014             NaN                 0.0

**Min Internet Users Percentage in 2014: Democratic People's Republic of Korea with 0%**

### Problem 20

In [116]:
gdp_int_2004[gdp_int_2004.Internet_Users_Pct == max(gdp_int_2004.Internet_Users_Pct)]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
5785  Sweden  2004    45029.814344               83.89

**Max Internet Users Percentage in 2004: Sweden with 83.89%**

In [117]:
gdp_int_2004[gdp_int_2004.Internet_Users_Pct == min(gdp_int_2004.Internet_Users_Pct)]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
6847  Dem. People's Rep. of Korea  2004             NaN                 0.0

**Min Internet Users Percentage in 2004: Democratic People's Republic of Korea with 0%**

In [118]:
gdp_int_2009[gdp_int_2009.Internet_Users_Pct == max(gdp_int_2009.Internet_Users_Pct)]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
2512  Iceland  2009    48428.312438                93.0

**Max Internet Users Percentage in 2009: Iceland with 93.00%**

In [119]:
gdp_int_2009[gdp_int_2009.Internet_Users_Pct == min(gdp_int_2009.Internet_Users_Pct)]

Country  Year  GDP_Per_Capita  Internet_Users_Pct
6842  Dem. People's Rep. of Korea  2009             NaN                 0.0

**Min Internet Users Percentage in 2009: Democratic People's Republic of Korea with 0%**

### Problem 21